src/conn/conn_sweep.c

```diff
@@ -45,6 +47,26 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /*
+         * When MongoDB opens cursors on table URIs, WiredTiger opens and releases table dhandles,
+         * particularly for simple tables. This repetitive dhandle creation and release, handled by
+         * the sweep server, leads to schema lock contention. To prevent this, the below algorithm
+         * modifies the sweep server's behavior. The algorithm now checks if any file dhandles are
+         * associated with the table before marking a table dhandle for sweeping, thus preventing
+         * unnecessary closures.
+         */
+        if (dhandle->type == WT_DHANDLE_TYPE_TABLE) {
+            table = (WT_TABLE *)dhandle;
+            if (table->is_simple) {
+                WT_WITHOUT_DHANDLE(session,
+                  WT_WITH_HANDLE_LIST_READ_LOCK(session,
+                    (ret = __wt_conn_dhandle_find(session, table->cgroups[0]->source, NULL))));
+
+                /* Continue if the file dhandle exists for the associated table dhandle. */
+                if (ret == 0)
+                    continue;
+            }
```

Comment:
Also, after checking ret against 0, it's probably good form to make sure that it doesn't have an error. (It really should be WT_NOTFOUND at this point and nothing else).   Normally we'd use something like `WT_RET_NOTFOUND_OK(ret)`, but this function has void return.  So maybe `WT_ASSERT_ALWAYS` to make sure it's not found.

Reply:
Fixed in 416ec0bb421e6ee88eb817e93ba55144633e9bfc